### PR TITLE
feature: creation of instrumented tests

### PR DESCRIPTION
**Pull request by luisoliveirazup**
Description: creation of instrumented tests of server driven screens with setId and toAndroidId

### CODE REVIEW

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        WaitHelper.waitForWithId(id)
+        onView(ViewMatchers.withId(id)).check(
+            matches(
+                isDisplayed()
+            )
+        )
+        return this
+    }
+
+    fun viewAndClick(id: Int): ScreenRobot {
```
Comment (FredericoNoce) [inline example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:48]: refactor method name

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.test.espresso.matcher.RootMatchers.isDialog
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import br.com.zup.beaglehackday.utils.WaitHelper
+import org.hamcrest.Matchers
+
+class ScreenRobot {
+
+    fun checkViewIdIsDisplayed(id: Int): ScreenRobot {
```
Comment (FredericoNoce) [inline example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:38]: refactor method name : checkViewIsDisplayed(id: Int)

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        onView(ViewMatchers.withId(id)).perform(ViewActions.click())
+        return this
+    }
+
+    fun clickOnTextDialog(text: String): ScreenRobot {
+        onView(withText(text)).inRoot(isDialog()).check(matches(isDisplayed())).perform(pressBack())
+        return this
+    }
+
+    fun checkOnText(text: String): ScreenRobot {
```
Comment (FredericoNoce) [inline example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:64]: refactor method name

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/utils/WaitHelper.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+object WaitHelper {
+    fun waitForWithId(id : Int){
+        var element : ViewInteraction
+        do{
+            element = Espresso.onView(ViewMatchers.withId(id))
+        } while (!MatcherExtension.exists(element))
+    }
+
+    fun waitForWithElement(element : ViewInteraction){
```
Comment (FredericoNoce) [inline example/android/app/src/androidTest/java/br/com/zup/beaglehackday/utils/WaitHelper.kt:38]: private ?

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt (hunk 1 above)
Comment (gabrielleonardozup) [reply example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:48]: checkViewIsDisplayedAndClickOnIt( ) maybe?

Comment (luisoliveirazup) [reply example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:48]: change done!

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt (hunk 3 above)
Comment (luisoliveirazup) [reply example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:64]: Change Done!

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt (hunk 2 above)
Comment (luisoliveirazup) [reply example/android/app/src/androidTest/java/br/com/zup/beaglehackday/robots/ScreenRobot.kt:38]: Change Done!

File: example/android/app/src/androidTest/java/br/com/zup/beaglehackday/utils/WaitHelper.kt (hunk 4 above)
Comment (luisoliveirazup) [reply example/android/app/src/androidTest/java/br/com/zup/beaglehackday/utils/WaitHelper.kt:38]: Excludes this method because it is not being used.

**Review by gabrielleonardozup — APPROVED** [review]

**Pull request merged**